### DISCUSSION TITLE
Dimension doesn't have time to load

### DISCUSSION THREAD

**Discussion by Timofffee** [Minestom/Minestom#1615] [Help] [answered]
Hello. I upgraded to the latest version of Minestom and immediately noticed that the lighting had changed. But that didn't fix my problem. My problem is that when a player enters an instance for the first time, it looks like dimension doesn't have time to load. Because of this, the lighting does not look as it should.
![image](https://user-images.githubusercontent.com/5556081/209376239-0fc93a17-a6c9-448c-8c28-c713a237e0ad.png)
However, if you die or fall out of the world, then the lighting becomes normal. 
![image](https://user-images.githubusercontent.com/5556081/209376172-c52544fa-15b9-4c33-92fd-2965dfd55156.png)

Any ideas how to fix it?

I also noticed that `.ambientLight(2.0)` no longer affects lighting.

**Comment by emortaldev** [comment]
Could you provide the full code where you are using the dimension? 

Also, ambientLight should be set to `1f` (it seems to work perfectly fine with values above 1f, but wiki.vg says 1f is max)

**Reply by Timofffee** [reply]
InstanceContainer constructor:
```
    DungeonInstance(String folder) {
        super(UUID.randomUUID(), Dimension.DARK);
        setTimeRate(0);
        setTime(18000);
        Path path = Path.of(folder);
        setChunkLoader(new AnvilLoader(folder));
    }
```
Dimentions
```
public class Dimension {
    public static final DimensionType DEFAULT = DimensionType.builder(NamespaceID.from("misttale:world"))
            .ambientLight(2.0F)
            .build();

    public static final DimensionType DARK = DimensionType.builder(NamespaceID.from("misttale:dark"))
            .skylightEnabled(true)
            .ambientLight(0.0F)
            .fixedTime(18000L)
            .logicalHeight(384)
            .height(384)
            .minY(-64)
            .build();

    static {
        MinecraftServer.getDimensionTypeManager().addDimension(DEFAULT);
        MinecraftServer.getDimensionTypeManager().addDimension(DARK);
    }
}
```

Create Instance and connect 
```
Instance instance = new DungeonInstance("test");
MinecraftServer.getInstanceManager().registerInstance(instance);
player.setInstance(instance, spawnPosition(player));
```

**Reply by Timofffee** [reply]
Now it looks not so much like a dimension error, but as a problem of transition between instances. The server creates a separate instance, but only partially moves the player there. That is, I see how the time has changed, how the territory has loaded and how events work. But as soon as I fall out of the world or die, then I have a screensaver loading the world.

**Reply by emortaldev** [reply]
Where is the bottom code? Like inside an event?

**Reply by emortaldev** [reply]
Also try registering the dimensions when the server starts or in an earlier event instead of in that static block

**Reply by Timofffee** [reply]
I tried to do this, but it did not give results. It seems that the server simply does not inform the client that the instance has a different dimension. The simplest version of the code I provided in issue #1619.

**Comment by Timofffee** [comment, marked as answer]
Fixed in #1619
